lib/Jsrt/JsrtInternal.h

```diff
@@ -207,23 +310,79 @@ JsErrorCode ContextAPINoScriptWrapper(Fn fn, bool allowInObjectBeforeCollectCall
             (scriptExceptionAllowed || errCode != JsErrorScriptException) &&
             errCode != JsErrorScriptTerminated);
     }
-    CATCH_STATIC_JAVASCRIPT_EXCEPTION_OBJECT
+    CATCH_STATIC_JAVASCRIPT_EXCEPTION_OBJECT(errCode)
 
     catch (const Js::JavascriptException& err)
     {
         AssertMsg(false, "Should never get JavascriptExceptionObject for ContextAPINoScriptWrapper.");
         scriptContext->GetThreadContext()->SetRecordedException(err.GetAndClear());
-        return JsErrorScriptException;
+        errCode = JsErrorScriptException;
     }
 
     catch (Js::ScriptAbortException)
     {
         Assert(scriptContext->GetThreadContext()->GetRecordedException() == nullptr);
         scriptContext->GetThreadContext()->SetRecordedException(scriptContext->GetThreadContext()->GetPendingTerminatedErrorObject());
-        return JsErrorScriptTerminated;
+        errCode = JsErrorScriptTerminated;
+    }
+
+    CATCH_OTHER_EXCEPTIONS(errCode)
+
+    return errCode;
+}
+
+template <class Fn>
+JsErrorCode ContextAPINoScriptWrapper_WithRecord(Fn fn, bool allowInObjectBeforeCollectCallback = false, bool scriptExceptionAllowed = false)
```

Comment:
Can we have record be a template parameter and combine this with the above function? They look somewhat identical